modules/rostests/kmtests/ntos_mm/ZwAllocateVirtualMemory.c

```diff
@@ -329,16 +368,51 @@ StressTesting(ULONG AllocationType)
     //free the allocated memory so that we can continue with the tests
     Status = STATUS_SUCCESS;
     Index = 0;
-    while (NT_SUCCESS(Status) && Index < RTL_NUMBER_OF(bases))
+    while (NT_SUCCESS(Status) && Index < NumberOfElements)
     {
         RegionSize = 0;
         Status = ZwFreeVirtualMemory(NtCurrentProcess(), &bases[Index], &RegionSize, MEM_RELEASE);
         bases[Index++] = NULL;
     }
 
+    ExFreePoolWithTag(bases, 'stts');
     return ReturnStatus;
 }
 
+static
+VOID
+ZeroBitsTest()
+{
+    NTSTATUS Status = STATUS_SUCCESS;
+    PVOID Base = NULL;
+    SIZE_T RegionSize = 1;
+    ULONG_PTR MaxPtr = -1;
+
+    /* https://stackoverflow.com/a/50553691 */
+#ifdef _WIN64
+    for (ULONG_PTR ZeroBits = MaxPtr; Status == STATUS_SUCCESS && ZeroBits > 32; ZeroBits >>= 1)
+    {
+        ZwAllocateVirtualMemory(NtCurrentProcess(), &Base, ZeroBits, &RegionSize, MEM_RESERVE | MEM_TOP_DOWN, PAGE_READWRITE);
+        trace("%p:%p - Status: %x\n", ZeroBits, Base, Status);
+        if ((ULONG_PTR)Base > ZeroBits)
+        {
+            ok_eq_pointer((ULONG_PTR)Base, (ZeroBits >> 16) << 16);
+        }
+        ZwFreeVirtualMemory(NtCurrentProcess(), &Base, &RegionSize, MEM_RELEASE);
+        Base = NULL;
+    }
+#endif
+
+    Status = STATUS_SUCCESS;
+    for (ULONG_PTR ZeroBits = 0; Status == STATUS_SUCCESS && ZeroBits <= 32; ZeroBits += 1)
```

Comment:
Not sure what the `Status` check inside the for-loop is, since you don't initialize Status to anything within the loop.
Shouldn't you do `Status = ZwAllocateVirtualMemory(...)` inside the loop?